<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="img_topics" default="[]"/>
    <arg name="camera_info_topics" default="[]"/>
    <arg name="hm_frame" default="base_link"/>
    <arg name="weights" default="$(dirname)/../config/weights/lss/lss.pt"/>
    <arg name="dphys_config_path" default="$(dirname)/../config/dphys_cfg.yaml"/>
    <arg name="lss_config_path" default="$(dirname)/../config/lss_cfg.yaml"/>
    <arg name="max_msgs_delay" default="0.1"/>
    <arg name="max_stale_msg_delay" default="0.2"/>
    <arg name="calib_path" default="''"/>
    <arg name="n_cams" default="''"/>

    <!-- Terrain encoder node (LSS) -->
    <node name="terrain_encoder" pkg="monoforce" type="terrain_encoder">
        <rosparam subst_value="true">
            dphys_config_path: $(arg dphys_config_path)
            lss_config_path: $(arg lss_config_path)
            img_topics: $(arg img_topics)
            camera_info_topics: $(arg camera_info_topics)
            hm_frame: $(arg hm_frame)
            weights: $(arg weights)
            max_msgs_delay: $(arg max_msgs_delay)
            max_stale_msg_delay: $(arg max_stale_msg_delay)
            calib_path: $(arg calib_path)
            n_cams: $(arg n_cams)
        </rosparam>
    </node>
</launch>
